### PR TITLE
[WIP] Feature/unit tests

### PR DESCRIPTION
**Pull request by irisdianauy**
Add unit tests to functions
Set up CI with Github Actions

### CODE REVIEW

File: .github/workflows/unittests_ci.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: Unit testing scanpy_scripts
+
+on: [push, pull_request]
+
```
Comment (pinin4fjords) [inline .github/workflows/unittests_ci.yaml:4]: Actually, could you add the testing to the existing suite? https://github.com/ebi-gene-expression-group/scanpy-scripts/blob/develop/.github/workflows/python-package.yml

You'll see that the dependencies etc are already in place there.

File: scanpy_scripts/click_utils.py [hunk 2]
```diff
@@ -36,6 +37,32 @@ def __init__(self, dtype=click.STRING, simplify=False, length=None):
...
+        Traceback (most recent call last):
+        ...
+        click.exceptions.BadParameter: 7.2 is not a valid integer
+        >>> test_cst2.convert('7', param, ctx)
+        Traceback (most recent call last):
+        ...
+        click.exceptions.BadParameter: 7 is not a valid comma separated list of length 2
+        >>> test_cst3.convert('7.2', param, ctx)
+        7.2
+        """
```
Comment (pinin4fjords) [inline scanpy_scripts/click_utils.py:65]: Nice set of tests- well done. I'd have been much more lazy ;-)

**Review by pinin4fjords — CHANGES_REQUESTED** [review]
In general I think these tests look very nice and comprehensive. If you can get the CI to pass this will be a great basis for addition of tests.

If possible, as you realised, there should be an over-arching functionality which discovers python files with doctests and runs them to execute those tests, rather than explicitly having to call every python file individually to run its tests.

File: .github/workflows/unittests_ci.yaml (hunk 1 above)
Comment (irisdianauy) [reply .github/workflows/unittests_ci.yaml]: Done in 0d4ad70cb2147286042854714504580eec446f7b.

File: .github/workflows/python-package.yml [hunk 3]
```diff
@@ -41,8 +41,14 @@ jobs:
         pip install -U setuptools>=40.1 wheel 'cmake<3.20'
         pip install $(pwd)/scanpy-scripts
         python -m pip install $(pwd)/scanpy --no-deps --ignore-installed -vv
+        pip install pytest
```
Comment (pinin4fjords) [inline .github/workflows/python-package.yml:44]: Just to be tidy, could you put these deps in the install statement on line 41?

**Review by pinin4fjords — APPROVED** [review]
Looking good - feel free to merge after addressing that very minor comment.

File: .github/workflows/python-package.yml (hunk 3 above)
Comment (irisdianauy) [reply .github/workflows/python-package.yml]: Dine in 378b115cbf1f64b1f4d5350469c95e77f0fec8fa

**Pull request merged**